Reland torch.ops API change machinery with the core functionality disabled

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71785

see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

anjali411 committed:
Reland torch.ops API change machinery with the core functionality disabled

see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

[ghstack-poisoned]

anjali411 committed:
Update on "Reland torch.ops API change machinery with the core functionality disabled"


see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33777229).

anjali411 committed:
Update on "Reland torch.ops API change machinery with the core functionality disabled"


see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33777229).

albanD commented on test/jit/test_custom_operators.py:
```diff
@@ -50,10 +50,6 @@ def test_default_arguments_are_used(self):
         output = torch.ops._test.leaky_relu(torch.tensor([-1.0, 1.0]))
         self.assertEqual(output, torch.tensor([-0.01, 1]))
 
-    def test_only_kwargs(self):
```
Why are these two tests removed?

albanD commented on test/test_per_overload_api.py:
```diff
@@ -0,0 +1,67 @@
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+class TestPerOverloadAPI(TestCase):
+    # def test_basics_opoverloadpacket(self):
```
nit: link to issue like in test_operators

anjali411 replied:
which issue?

albanD replied:
The same as the one you link in test_operators above? I supposed it was an issue tracking enabling this feature once we can fix the internal issues?

anjali411 replied:
no it's a different issue. I don't think we need to link any issue right now Alban. We'll enable it by next week, if all goes well.

albanD replied:
Sounds good to me!

albanD commented on torch/_ops.py:
```diff
@@ -59,13 +162,20 @@ def __getattr__(self, op_name):
         # with qualified_op_name
         torch.jit._builtins._register_builtin(op, qualified_op_name)
+        op.__module__ = self.__module__ + "." + namespace_name
+        # opoverloadpacket = OpOverloadPacket(qualified_op_name, op_name, op)
```
link to the issue here as well

albanD commented on torch/jit/_script.py:
```diff
@@ -1270,7 +1270,6 @@ def forward(self, a) -> MyModule:
         return create_script_dict(obj)
     if isinstance(obj, list):
         return create_script_list(obj)
-
```
spurious line change

albanD reviewed (commented):
Only minor things

anjali411 committed:
Update on "Reland torch.ops API change machinery with the core functionality disabled"


see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

[ghstack-poisoned]

anjali411 committed:
Update on "Reland torch.ops API change machinery with the core functionality disabled"


see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

[ghstack-poisoned]

albanD reviewed (approved):
SGTM
